src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -104,6 +114,20 @@ public GoogleProjectResource getProjectResourceById(UUID id) {
         return resourceDao.retrieveProjectById(id);
     }
 
+    public List<UUID> markUnusedProjectsForDelete(String profileId) {
+        return resourceDao.markUnusedProjectsForDelete(UUID.fromString(profileId));
+    }
+
+    public void deleteUnusedProjects(List<UUID> projectIdList) {
+        for (UUID projectId : projectIdList) {
+            deleteProjectResource(projectId);
+        }
+    }
+
+    public void deleteProjectMetadata(List<UUID> projectIdList) {
+        resourceDao.deleteProjectMetadata(projectIdList);
+    }
```

Comment:
If I'm understanding correctly, there are two levels of delete here:
1. Deleting the actual google project from gcloud
2. Deleting the entry for the project from the project_resource table

I've found our naming conventions for these methods confusing. For example, on L123, we're deleting the actual project (item 1), but we're calling it "deleteProject**Resource**." We also call it "deleteGoogleProject" in the GoogleProjectService.  [Elsewhere in the code](https://github.com/DataBiosphere/jade-data-repo/pull/698/files#diff-3c317ebf6e0dec044f6b8dc873989c364559056636b5fc89dab98a84829b6aadR127), we reference "deleteProject" as just deleting the entry in the table (item 2). 

I think it could be helpful to more clearly define/name these methods, and clarify this in your comments.
Potential naming conventions:
1. Delete Google Project (or delete project resource?)- Deleting the actual google project from gcloud
2. Delete Project Metadata - Deleting the entry for the project in the project_resource table

Reply:
Really good point. One of the naming things that confuses me is the class name "GoogleBucketResource". It is easy to read that as a Google object, when in fact it is our object. OTOH, when we have an Azure package, we will have an AzureBucketResource. So I left Jeremy's class naming intact.

I renamed a some things to try to make it clearer. Now there are two methods for `deleteGoogleProject` - one that takes the `googleProjectId` and the other that takes `metadataProjectId`. I think that helps clear up the weirdness of `deleteUnusedProjects` calling something that sounds like it should delete metadata as well as data.